File: src/ads_runner_client/lib/domain/entities/screen_device.dart
import 'package:equatable/equatable.dart';
import '../../core/enums/screen_status.dart';

class ScreenDevice extends Equatable {
  final String id;
  final String name;
  final String vehiclePlate;
  final double latitude;
  final double longitude;
  final ScreenStatus status;
  final DateTime lastSeen;
  final String firmwareVersion;
  final List<String> assignedCampaignIds;

  const ScreenDevice({
    required this.id,
    required this.name,
    required this.vehiclePlate,
    required this.latitude,
    required this.longitude,
    required this.status,
    required this.lastSeen,
    required this.firmwareVersion,
    required this.assignedCampaignIds,
  });

  @override
  List<Object> get props => [id, name, vehiclePlate, status];
}

File: src/AdsRunner.Domain/Entities/ScreenDevice.cs
using AdsRunner.Domain.Common;
using AdsRunner.Domain.Enums;

namespace AdsRunner.Domain.Entities;

public class ScreenDevice : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string VehiclePlate { get; set; } = string.Empty;
    public double Latitude { get; set; }
    public double Longitude { get; set; }
    public ScreenStatus Status { get; set; }
    public DateTime LastSeen { get; set; }
    public string FirmwareVersion { get; set; } = string.Empty;

    // Navigation properties
    public ICollection<CampaignScreen> CampaignScreens { get; set; } = new List<CampaignScreen>();
}
